### PR TITLE
Add join-by-url

### PR DESCRIPTION
**Pull request by ijt**
This feature allows users to join a community by visiting URLs of the form

http://hylo.com/c/:community/join/:code

### CODE REVIEW

**Comment by edwardwest** [comment]
awesome--  thanks, Issac!

_______________________
*Edward West*
mobile 415.225.7335
Facebook <http://www.facebook.com/edwardjwest>  Twitter
<http://www.twitter.com/edwardwest>  LinkedIn
<http://www.linkedin.com/in/edwardwest>

File: src/js/app/controllers/community/JoinCommunityByUrlCtrl.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = function($scope, $controller, community, code) {
+  'ngInject';
+
+  $scope.community = community;
+  $scope.code = code;
+
+  $controller('JoinCommunityCtrl', {$scope: $scope})
```
Comment (levity) [inline src/js/app/controllers/community/JoinCommunityByUrlCtrl.js]: what does this do?

**Comment by levity** [comment]
Does this work seamlessly if you're not signed up yet?

File: src/js/app/controllers/community/JoinCommunityByUrlCtrl.js (hunk 1 above)
Comment (ijt) [inline src/js/app/controllers/community/JoinCommunityByUrlCtrl.js]: This puts all the variables and functions from JoinCommunityCtrl onto $scope. It's like subclassing JoinCommunityByUrlCtrl from JoinCommunityCtrl.

**Comment by ijt** [comment]
Not yet. I'll add that now.

**Comment by edwardwest** [comment]
also-- could we surface a line in moderator settings underneath the
invitation code section as follows?:

*Invitation code*
foo-invite-code

This code can be given to people to allow them to join the community,
instead of sending individual invitations by email.


*Invitation code link*
You can copy this link for pasting in emails or embedding on your webpage
to pre-populate the invite code for new members to easily join.

www.hylo.com/c/foo-community/join-foo-invite-code=foo-invite-code
 [button->Copy Link to Clipboard]



_______________________
*Edward West*
mobile 415.225.7335
Facebook <http://www.facebook.com/edwardjwest>  Twitter
<http://www.twitter.com/edwardwest>  LinkedIn
<http://www.linkedin.com/in/edwardwest>

**Comment by levity** [comment]
We're already surfacing the invitation code

**Comment by edwardwest** [comment]
I know-- I just wanted Issac to have context for where it should go in the
options

_______________________
*Edward West*
mobile 415.225.7335
Facebook <http://www.facebook.com/edwardjwest>  Twitter
<http://www.twitter.com/edwardwest>  LinkedIn
<http://www.linkedin.com/in/edwardwest>

**Comment by levity** [comment]
ah, gotcha

**Comment by ijt** [comment]
Got it. Sure.

**Comment by edwardwest** [comment]
awesome!

On Tuesday, October 6, 2015, Issac Trotts <notifications@github.com> wrote:

> Got it. Sure.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Hylozoic/hylo-frontend/pull/63#issuecomment-146046168>
> .
>


-- 
_______________________
*Edward West*
mobile 415.225.7335
Facebook <http://www.facebook.com/edwardjwest>  Twitter
<http://www.twitter.com/edwardwest>  LinkedIn
<http://www.linkedin.com/in/edwardwest>

**Comment by ijt** [comment]
I'm heading out for a jog. Will add the settings update when I get back.

**Comment by levity** [comment]
replaced by #65

**Pull request closed without merging**